### PR TITLE
[main extended] Fix dnf-plugins-core, ocaml-findlib builds

### PR DESCRIPTION
**Pull request by thcrain-msft**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Fix builds of extended specs dnf-plugins-core, ocaml-findlib

###### Change Log  

- ocaml-findlib: Add patch for detection of ocaml stdlib features in ocaml 4.13.X
- dnf-plugins-core: Only ask for python3-nose during test builds

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Local package builds

### CODE REVIEW

File: SPECS-EXTENDED/dnf-plugins-core/dnf-plugins-core.spec [hunk 1]
```diff
@@ -11,7 +11,7 @@ Distribution:   Mariner
 
 Name:           dnf-plugins-core
 Version:        4.0.18
-Release:        4%{?dist}
+Release:        15%{?dist}
```
Comment (jslobodzian) [inline SPECS-EXTENDED/dnf-plugins-core/dnf-plugins-core.spec:14]: I think you mean 5 here, not 15.

File: SPECS-EXTENDED/hfsplus-tools/hfsplus-tools.spec [hunk 2]
```diff
@@ -2,7 +2,7 @@ Vendor:         Microsoft Corporation
 Distribution:   Mariner
 Name:           hfsplus-tools
 Version:        540.1.linux3
```
Comment (jslobodzian) [inline SPECS-EXTENDED/hfsplus-tools/hfsplus-tools.spec:4]: is "linux3" the actual version number?

Comment (thcrain-msft) [reply SPECS-EXTENDED/hfsplus-tools/hfsplus-tools.spec:4]: Yep, this is part of the version as shipped by upstream.

**Comment by thcrain-msft** [comment]
Reverting hfsplus-tools package- no available upstream

**Review by jslobodzian — APPROVED** [review]

**Review by jslobodzian — APPROVED** [review]

**Review by dmcilvaney — APPROVED** [review]

**Pull request merged**